File: sed_hsf_der.cpp
#include "sed_hsf_der.h"

double sed_hsf_der(double phi)
{
	struct test_cases* pt_test = get_tests();

    if (phi < 0 || phi > pt_test->phimax)
        return 0;
    else if (phi > pt_test->phi0)
    {
        double vphi0 = pow(1 - pt_test->phi0, pt_test->nexp - 2);
        double v1phi0 = -(pt_test->nexp - 2)*pow(1 - pt_test->phi0, pt_test->nexp - 3);
        return pt_test->vinf*(-vphi0 + v1phi0*(1 + pt_test->phi0) - 2*phi*v1phi0);
    }
    else
        return -(pt_test->nexp - 1)*pt_test->vinf*pow(1 - phi, pt_test->nexp - 2);
}

File: sed_hsf_der.m
function v=sed_hsf_der(phi)

%double: phi, v, vinf, nexp, phimax, phi0, vphi0, v1phi0

global vinf phimax phi0 nexp;

if (phi < 0 || phi > phimax)
    v = 0;
elseif (phi > phi0)
    vphi0 = (1 - phi0)^(nexp - 2);
    v1phi0 = -(nexp-2)*(1 - phi0)^(nexp - 3);
    v = vinf*(-vphi0 + v1phi0*(1 + phi0) - 2*phi*v1phi0);
else
    v = -(nexp - 1)*vinf*(1 - phi)^(nexp - 2);
end